shared/common-adapters/avatar.desktop.js

```diff
@@ -14,26 +17,54 @@ type State = {
 
 class Avatar extends PureComponent<void, Props, State> {
   state: State;
+  _onURLLoaded: ?() => void;
 
   constructor (props: Props) {
     super(props)
     this.state = {avatarLoaded: false, errored: false}
   }
 
+  componentDidMount () {
+    this._onURLLoaded = () => { this.forceUpdate() }
+  }
+
+  componentWillUnmount () {
+    this._onURLLoaded = null
+  }
+
   componentWillReceiveProps (nextProps: Props) {
     const url = shared.createAvatarUrl(this.props)
     const nextUrl = shared.createAvatarUrl(nextProps)
 
     if (url !== nextUrl) {
       this.setState({avatarLoaded: false, errored: false})
+      // $FlowIssue
+      if (this.props.username && _urlLookup) {
+        _urlLookup(this.props.username, () => {
+          this._onURLLoaded && this._onURLLoaded()
+        })
+      }
     }
   }
 
   render () {
-    const {size} = this.props
+    // $FlowIssue
+    const {size, url, username} = this.props
     const width = size
     const height = size
-    const url = shared.createAvatarUrl(this.props) || noAvatar
+    let resolvedURL
+    if (url) {
+      resolvedURL = url
+    } else if (username) {
+      resolvedURL = _urlLookup(username, () => {
```

Comment:
Would be better practice to move this outside of the `render` call since it has side-effects. Here is an example of the idiomatic way to implement this (see the last example): https://github.com/facebook/react/issues/7671#issue-175393882